src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -49,15 +45,14 @@ template <typename DeviceType>
 struct DistributedSearchTreeImpl
 {
   // spatial queries
-  template <typename ExecutionSpace, typename Predicates>
-  static void
-  queryDispatch(SpatialPredicateTag,
-                DistributedSearchTree<typename DeviceType::memory_space,
-                                      void> const &tree,
+  template <typename DistributedTree, typename ExecutionSpace,
+            typename Predicates, typename Indices, typename Offset,
+            typename Ranks>
+  static std::enable_if_t<Kokkos::is_view<Indices>{} &&
+                          Kokkos::is_view<Offset>{} && Kokkos::is_view<Ranks>{}>
+  queryDispatch(SpatialPredicateTag, DistributedTree const &tree,
```

Comment:
Do you want to allow any other `DistributedTree` in this pull request?
Otherwise, I would rather use
```suggestion
  queryDispatch(SpatialPredicateTag, DistributedSearchTree<MemorySpace, void> const &tree,
```
(here and below).

Reply:
I wanted to get rid of the forward declaration.  Also this is one less place to update if we add template parameters to the distributed tree.